CryptoPkg/Library/BaseCryptLibMbedTls/Pk/CryptPkcs7VerifyCommon.c

```diff
@@ -299,42 +333,73 @@ Pkcs7GetSignedData (
       &P, End, &SignedData->DigestAlgorithms);
   }
 
-  if (Ret == 0 &&
-    (SignedData->DigestAlgorithms.len != sizeof (MBEDTLS_OID_DIGEST_ALG_SHA256) - 1)) {
-    Ret = -1;
-  }
-
-  if (Ret == 0 &&
-    CompareMem (
-      SignedData->DigestAlgorithms.p,
-      MBEDTLS_OID_DIGEST_ALG_SHA256,
-      SignedData->DigestAlgorithms.len) != 0){
-    Ret = -1;
+  if (Ret == 0) {
+    if (((SignedData->DigestAlgorithms.len == sizeof (MBEDTLS_OID_DIGEST_ALG_SHA1) - 1) &&

```

Comment:
why you still have SHA1? Is this absolutely needed?

Reply:
Thanks. I have deleted the SHA1 in Pkcs7.